### PR TITLE
Update JAX and fix aarch64-darwin build

### PR DESCRIPTION
**Pull request by samuela**
###### Description of changes

This PR does a few things:
* Update jaxlib(-bin): 0.3.22 -> 0.4.4
* Update jax: 0.4.1 -> 0.4.5
* Fix the `rev` that JAX was building on which was messed up by https://github.com/NixOS/nixpkgs/commit/0d6a07172dacc46df0a72487a8931c9d4ba2c0dc which was merged without approval.
* Unbreak JAX on aarch64-darwin. The `jaxlib` source build still doesn't work on this platform but that doesn't mean we can use the `jaxlib-bin` package in its place, esp. considering it's only a `checkPhase` dependency of `jax`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samuela** [comment]
There are a few TODOs before this can be merged:
- [ ] jaxlib is currently failing to build on x86_64-linux due to https://github.com/google/jax/issues/14795.
- [ ] We'll need the `fetchAttrs` hashes for platforms that I do not have access to: https://github.com/NixOS/nixpkgs/pull/219778/files#diff-8906f9e778af6b3dc2ab4057b61c97ee82f0f35b3d02f49a5beabb4de4d64c23R252-R253. @divanorama previously updated these hashes, but I only have access to x86_64-linux and aarch64-darwin machines. I'm leaning towards (temporarily?) dropping support for the other platforms since I have no way of testing/building on them as a maintainer. @uri-canva IIRC you previously worked on the x86_64-darwin builds. Do you still rely on these builds? Would you be able to test builds on that platform?

**Comment by samuela** [comment]
I should add that although we may drop support for from-source `jaxlib` builds for platforms, they don't necessarily need to be dropped from `jaxlib-bin` since that's easier to maintain. So support would not be dropped entirely.

**Comment by samuela** [comment]
This PR should be ready to go now. The JAX team was kind enough to hook us up with a new snappy release in order to fix the build (s/o @hawkinsp! 🚀)

I've tested that jaxlib builds with and without CUDA support. I'm running a nixpkgs-review now...

**Comment by samuela** [comment]
cc @NixOS/cuda-maintainers @uri-canva @ndl

**Comment by samuela** [comment]
If we don't have anyone with access to a machine to test it out conveniently I vote that we drop support from the source build and x86_64-darwin users can use the jaxlib-bin package instead. Otherwise the maintenance load is just too high IMHO.

**Comment by SomeoneSerge** [comment]
Feels reasonable. For the sake of documentation,  I suggest that if one later wants to reintroduce x8664Darwin support they also set up a GitHub Actions workflow that runs over a matrix of platforms and tries to update the derivation (version and hashes) via something like nix-update. And I don't expect nix-update to just work here. The whole business of conditional hashes is too bearing

**Comment by samuela** [comment]
> We could also fix it so the fetch hashes are not system dependent, though I'm not sure how hard it would be. I'll take a look. Are there any other reasons to keep jax both built from source and as a binary?

jaxlib-bin comes in handy since the builds are so simple and fast. So eg that makes it a lot easier to support more architectures and lets users more easily override to get different versions of jaxlib if necessary. The full jaxlib source builds are just so looooooong

**Comment by samuela** [comment]
snappy is fairly popular, so this actually results in a few hundred rebuilds. I was hoping to run a nixpkgs-review and be done with it, but I actually ran out of disk space on my 200Gb drive. Would someone else be able to nixpkgs-review it? If not, I could create a snappy-only PR targeting staging.

**Comment by samuela** [comment]
Spun off the snappy change into https://github.com/NixOS/nixpkgs/pull/221215. @uri-canva could you ptal?

We should be able to rebase and merge this once that chance lands on master.

**Comment by samuela** [comment]
> but I'm having trouble getting the diff of the fetch derivation, I'll have to come back to it another time.

wdym? it built fine for me on aarch64-darwin

> Are there things the source build supports that the binary doesn't? I'm confused why we need both, I thought we had both as a workaround for some issue that prevented us from building from source, but it doesn't sound like it's a transitional thing. Do we still need both?

Are you proposing removing the source build or removing the binary build? i think the main selling point for the source build is just that nixpkgs generally prefers source builds 🤷

**Comment by samuela** [comment]
> Yeah but I want to build it on a different system too so I can diff them. Usually if different systems need different shas it means the deps have system specific files in them, which we can delete and rebuild during the build. Ideally the output of the fetch only has system independent files that were downloaded.

Ohhhh this is in regards to `fetchAttrs`? Yeah I don't really know anything about that. It'd be really great if we could get all platforms on the same `fetchAttrs` hash!

> I'm proposing keeping only one of the two, so we don't duplicate the maintenance effort. Ideally it would be the source one, that's why I'm trying to figure out what is causing the diff in the deps shas, so we can develop the derivation for all systems more easily.

There are still benefits to having the binary build around. It's a lot easier to hack on, and users (like myself) occasionally need to override things in order to upgrade/downgrade versions, play around with settings, etc. Luckily the maintenance burden of the binary build is much lower.

**Comment by samuela** [comment]
> Right then maybe we should keep only the binary build?

I would be sympathetic to such a change, but IIUC there are a number of people who really like to have the source build around

**Comment by divanorama** [comment]
Fwiw, quickly checked a diff in `fetchAttr` between `{linux,darwin}-x86_64` (`/nix/store/b7f4k0i0pl95fxdlxh2q6a4plr6rn4k3-bazel-build-jaxlib-0.3.22-deps.tar.gz` and `/nix/store/9yl3a0nn2wk9ps8l0wps314msy0z8vja-bazel-build-jaxlib-0.3.22-deps.tar.gz`)
There's some diff
```diff
diff -r a/external/nsync/WORKSPACE b/external/nsync/WORKSPACE
1,2d0
< # DO NOT EDIT: automatically generated WORKSPACE file for _tf_http_archive rule
< workspace(name = "nsync")
...
Only in b/external/nsync: bazel
Only in b/external/nsync: BUILD
Only in a/external/nsync: BUILD.bazel
Only in b/external/nsync: builds
Only in b/external/nsync: CMakeLists.txt
Only in b/external/nsync: CONTRIBUTING
Only in b/external/nsync: .gitignore
Only in b/external/nsync: internal
Only in b/external/nsync: LICENSE
Only in b/external/nsync: platform
Only in b/external/nsync: public
Only in b/external/nsync: README
Only in b/external/nsync: testing
Only in b/external/nsync: tools
```
there are also bunch of broken links in archives, not sure if they get unbroken during real build
```console
$ ls -la a/external/nsync
... BUILD.bazel -> NIX_BUILD_TOP/output/external/org_tensorflow/third_party/systemlibs/nsync.BUILD
... WORKSPACE
```

**Comment by samuela** [comment]
I _think_ we can safely ignore that since we're not using their vendored nsync version and instead providing our own. Not sure how easy that is to rectify.

One thing I'm curious about is the diff b/w cuda and non-cuda fetchAttrs versions... it's not clear to me what the difference would be between them

**Comment by divanorama** [comment]
Cuda `/nix/store/64makcssc5zfg06i64wi4sdhhpwzg62m-bazel-build-jaxlib-0.3.22-deps.tar.gz` as `c` compared to linux-x86_64
```
Only in c/external: cudnn_frontend_archive
Only in c/external: @cudnn_frontend_archive.marker
Only in c/external: nccl_archive
Only in c/external: @nccl_archive.marker
```

**Comment by samuela** [comment]
huh i wonder what cudnn_frontend_archive is. we should be using our own nccl version, so i don't think that one should matter

**Comment by divanorama** [comment]
That's `nccl_archive` info
```
# /build/source/WORKSPACE:38:14
_tf_http_archive(
  name = "nccl_archive",
  generator_name = "nccl_archive",
  generator_function = "workspace",
  sha256 = "d5f5243200d4e40683c56f04435bfd6defa379cb4f2b8c07b0f191df0f66c3d9",
  urls = ["https://storage.googleapis.com/mirror.tensorflow.org/github.com/nvidia/nccl/archive/v2.13.4-1.tar.gz", "https://github.com/nvidia/nccl/archive/v2.13.4-1.tar.gz"],
  strip_prefix = "nccl-2.13.4-1",
  patch_file = ["//third_party/nccl:archive.patch"],
  build_file = "//third_party:nccl/archive.BUILD",
)
# Rule nccl_archive instantiated at (most recent call last):
#   /build/source/WORKSPACE:38:14                                                                                                         in <toplevel>
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/tensorflow/workspace2.bzl:938:21 in workspace
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/tensorflow/workspace2.bzl:599:20 in _tf_repositories
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/third_party/repo.bzl:136:21      in tf_http_archive
# Rule _tf_http_archive defined at (most recent call last):
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/third_party/repo.bzl:89:35 in <toplevel>
```
and for `cudnn_frontend_archive`
```
# /build/source/WORKSPACE:38:14
_tf_http_archive(
  name = "cudnn_frontend_archive",
  generator_name = "cudnn_frontend_archive",
  generator_function = "workspace",
  sha256 = "6ca6e7d4affdff59c749865d6d0428c849968b0873a1d1b849f56d7be624f27b",
  urls = ["https://storage.googleapis.com/mirror.tensorflow.org/github.com/NVIDIA/cudnn-frontend/archive/refs/tags/v0.7.1.zip", "https://github.com/NVIDIA/cudnn-frontend/archive/refs/tags/v0.7.1.zip"],
  strip_prefix = "cudnn-frontend-0.7.1",
  patch_file = ["//third_party:cudnn_frontend_header_fix.patch"],
  build_file = "//third_party:cudnn_frontend.BUILD",
)
# Rule cudnn_frontend_archive instantiated at (most recent call last):
#   /build/source/WORKSPACE:38:14                                                                                                         in <toplevel>
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/tensorflow/workspace2.bzl:938:21 in workspace
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/tensorflow/workspace2.bzl:169:20 in _tf_repositories
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/third_party/repo.bzl:136:21      in tf_http_archive
# Rule _tf_http_archive defined at (most recent call last):
#   /build/.cache/bazel/_bazel_homeless-shelter/664ed89c9c1572b2edd071f7898c14a5/external/org_tensorflow/third_party/repo.bzl:89:35 in <toplevel>
```
looks like they come from tensorflow repo and either added conditionally on whether cuda is enabled, or maybe only used conditionally (didn't check if they'd appear in build directory if declared by not used).

Another thing I didn't investigate yet is why `deps` archive creation seems to be calling `bazel build`, sounds excessive. But maybe this way it can prune out unused external dependencies?

**Comment by samuela** [comment]
snappy 1.1.10 has now left staging and landed onto master. I'll rebase this PR and we should be good to go.

**Comment by samuela** [comment]
just rebased, removing the now-redundant snappy commit, and happy to report that everything is building as expected on x86_64-linux. @uri-canva currently trying your "overfetch" trick from https://github.com/NixOS/nixpkgs/pull/221390 to see if we can get that landed as well.

Since this commit there jaxlib 0.4.7 and jax 0.4.8 have been released but I vote that we keep those upgrades to a future PR.

**Comment by samuela** [comment]
Ok I'll go ahead and merge tomorrow if no one objects

File: pkgs/development/python-modules/jaxlib/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/jaxlib/default.nix:243]: Oh God, are we really rid of this? 😍

**Review by SomeoneSerge — APPROVED** [review]

File: pkgs/development/python-modules/jaxlib/default.nix
Comment (samuela) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: yessir!! s/o to @uri-canva !

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: Ouch. I guess we'll need to a nixpkgs-review then

Comment (samuela) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: On x86_64-linux the `jaxlib`, `jaxlibWithCuda`, and `jax` derivations all built fine for me. (I did have to fiddle with the hash)

Since aarch64-darwin has never been supported by the source build, and x86_64-darwin support is waning, presumed to be broken now, I think x86-64_linux is the only platform that we're really supporting the source build on atm. Do we really need a nixpkgs-review then?

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: ...well, it offloads the mental load to the machine

Comment (samuela) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: Ah @uri-canva you are correct: ofborg is failing https://gist.github.com/GrahamcOfBorg/819dce99401b0659eb259dd2e6d6e752

Any objections to marking aarch64-linux as broken? Or perhaps removing it from `meta.platforms`?

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: I'm a bit lost now, why is "attribute 'aarch64-linux' missing", and was it missing before?

Comment (samuela) [reply pkgs/development/python-modules/jaxlib/default.nix:243]: On a second look, we are already specifying `meta.platforms` correctly

https://github.com/NixOS/nixpkgs/blob/37e8b04cc76a2af79994b2b4439bf67e00559082/pkgs/development/python-modules/jaxlib/bin.nix#L128

But for some reason ofborg is still trying to evaluate the derivation. I opened [a discussion on Discourse](https://discourse.nixos.org/t/ofborg-does-not-respect-meta-platforms/27019) to track.

**Comment by samuela** [comment]
Note to self: We should prob merge https://github.com/NixOS/nixpkgs/pull/221390 before this

**Comment by GaetanLepage** [comment]
Thank you for this updating effort !
I would suggest that we directly go to jaxlib v0.4.7 and jax v0.4.8.

I think that we might need to change the `prefetch.sh` script as is:
```
version="$1"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/nocuda/jaxlib-${version}-cp310-cp310-manylinux2014_x86_64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/nocuda/jaxlib-${version}-cp311-cp311-manylinux2014_x86_64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/mac/jaxlib-${version}-cp310-cp310-macosx_10_14_x86_64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/mac/jaxlib-${version}-cp311-cp311-macosx_10_14_x86_64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/mac/jaxlib-${version}-cp310-cp310-macosx_11_0_arm64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/mac/jaxlib-${version}-cp311-cp311-macosx_11_0_arm64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/cuda12/jaxlib-${version}+cuda12.cudnn88-cp310-cp310-manylinux2014_x86_64.whl)"
nix hash to-sri --type sha256 "$(nix-prefetch-url https://storage.googleapis.com/jax-releases/cuda12/jaxlib-${version}+cuda12.cudnn88-cp311-cp311-manylinux2014_x86_64.whl)"
```

File: pkgs/development/python-modules/jaxlib/bin.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/jaxlib/bin.nix:52]: Should we also account for `pythonVersion`?

**Comment by samuela** [comment]
Every time I try to run nixpkgs-review it hangs, apparently on the cvxpy test suite. Not sure why as the cvxpy build does not seem to hang either on this branch or on master.

**Comment by SomeoneSerge** [comment]
> Every time I try to run nixpkgs-review it hangs, apparently on the cvxpy test suite. Not sure why as the cvxpy build does not seem to hang either on this branch or on master.

I've been continuously suffering from cvxpy tests when running parallel builds, and I haven't quite figured out why. Upstream github actions reveal that their tests take at most two minutes

**Comment by GaetanLepage** [comment]
> Every time I try to run nixpkgs-review it hangs, apparently on the cvxpy test suite. Not sure why as the cvxpy build does not seem to hang either on this branch or on master.

Could you try to update this branch ? I can give it a shot on my machine if you want.

**Comment by SomeoneSerge** [comment]
@samuela if you get bored, I think it's OK to

```
❯ sudo kill -9 $(pgrep -f cvxpy)
```

The failures we're looking for are likely unrelated to cvxpy...

**Comment by samuela** [comment]
> I've been continuously suffering from cvxpy tests when running parallel builds, and I haven't quite figured out why. Upstream github actions reveal that their tests take at most two minutes

Glad to hear I'm not alone here!

> If you get bored, I think it's OK to [...]

Brilliant! I'll see how far I can get with that. I really wish nixpkgs-review had an option to skip packages...

**Comment by GaetanLepage** [comment]
@samuela When trying to build jaxlib 0.4.7 from source I faced a build failure at the `preBuild` phase.
More precisely, linking the protobuf lib in `../output/external/org_tensorflow` was failing because this folder doesn't exist.
I removed this patch and it built succesfully.

It seems to come from [this upstream commit](https://github.com/google/jax/commit/172a831219aa7d3524c0c8b5779dc29597a05810).

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 219778` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>43 packages failed to build:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pytensor</li>
    <li>python310Packages.pytensor.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.dm-haiku.testsout</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
  </ul>
</details>
<details>
  <summary>45 packages built:</summary>
  <ul>
    <li>python310Packages.aeppl</li>
    <li>python310Packages.aeppl.dist</li>
    <li>python310Packages.aesara</li>
    <li>python310Packages.aesara.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.chex</li>
    <li>python310Packages.chex.dist</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-haiku.dist</li>
    <li>python310Packages.dm-haiku.testsout</li>
    <li>python310Packages.jax</li>
    <li>python310Packages.jax.dist</li>
    <li>python310Packages.jaxlib (python310Packages.jaxlib-build ,python310Packages.jaxlibWithoutCuda)</li>
    <li>python310Packages.jaxlib-bin</li>
    <li>python310Packages.jaxlib-bin.dist</li>
    <li>python310Packages.jaxlib.dist (python310Packages.jaxlib-build.dist ,python310Packages.jaxlibWithoutCuda.dist)</li>
    <li>python310Packages.jaxlibWithCuda</li>
    <li>python310Packages.jaxlibWithCuda.dist</li>
    <li>python310Packages.jmp</li>
    <li>python310Packages.jmp.dist</li>
    <li>python310Packages.objax</li>
    <li>python310Packages.objax.dist</li>
    <li>python310Packages.optax</li>
    <li>python310Packages.optax.dist</li>
    <li>python310Packages.optax.testsout</li>
    <li>python310Packages.tensorflow-bin</li>
    <li>python310Packages.tensorflow-bin.dist</li>
    <li>python310Packages.treeo</li>
    <li>python310Packages.treeo.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
  </ul>
</details>

**Comment by samuela** [comment]
###### Failed derivations

- [ ] tensorflow_probability-0.19.0-py2.py3-none-any.whl.drv (derivation hash: m5hnwrjlfw35c8m12glr6f5vhqs47g1l)
  - https://gist.github.com/samuela/764dac33adcc2c3cfe20050af6cfe125
- [ ] python3.10-flax-0.6.5.drv (derivation hash: 4nzqqvf0ni6yfcv8b5gf7wqfdsd1qwrh)
  - https://gist.github.com/samuela/78fe1a8c60b527583d6384cdaaa16679
- [ ] python3.11-jmp-unstable-2021-10-03.drv (derivation hash: bqn5sdpw8fll1y0fz3pdkdzz72b14p1a)
  - https://gist.github.com/samuela/2fca6aa1344494ed52d60f39b3261749
- [ ] python3.10-tensorflow-datasets-4.8.2.drv (derivation hash: rmw0l6014xm0pkyskm88wlzb956r7fy6)
  - https://gist.github.com/samuela/8d00e41b98fe7351ed11473e6c7b22ef
- [ ] python3.10-cvxpy-1.3.0.drv (derivation hash: n2ig5g2fbvgrirqcnmmvba0pz1fj2kw3)
  - https://gist.github.com/samuela/3a45e122d31de4139c108d9deedbb588
- [ ] python3.11-cvxpy-1.3.0.drv (derivation hash: xyyrmk6kxpy1i05f6vyca7g48wfr2vb9)
  - https://gist.github.com/samuela/1a012a96311e79847292bd45031202c1
- [ ] python3.10-pytensor-2.10.1.drv (derivation hash: pvnd0l1vjrnxvnmj4ssf4c5laa469zay)
  - https://gist.github.com/samuela/dcaafc5bc8319550513fed6332a5a18e
- [ ] python3.11-numpyro-0.11.0.drv (derivation hash: 6mb3rzixdwr4nr0s8rdhdc4rrck7fsk0)
  - https://gist.github.com/samuela/f33cd1c0be79dce552809863f62e9471
- [ ] python3.10-numpyro-0.11.0.drv (derivation hash: 1piw5qwr4lswd04lw3aglr302704dbdp)
  - https://gist.github.com/samuela/d1ca6da8b6aa43f466154a9764ee4e24

**Comment by samuela** [comment]
Doesn't look like any of these failures are JAX version related AFAICT. I propose we go ahead and merge this and then address the python version-specific downloads and 0.4.7/8 upgrade in separate PR(s).

**Review by uri-canva — APPROVED** [review]

**Comment by SomeoneSerge** [comment]
We're still at `80|   src = if !cudaSupport then cpuSrcs."${stdenv.hostPlatform.system}" else gpuSrc;`?

**Comment by samuela** [comment]
> We're still at 80|   src = if !cudaSupport then cpuSrcs."${stdenv.hostPlatform.system}" else gpuSrc;?

You mean we're not yet branching on the python version? Or something else?

**Comment by SomeoneSerge** [comment]
> You mean we're not yet branching on the python version? Or something else?

No, the ofborg thing?

**Comment by samuela** [comment]
Ohhh, I totally forgot about that! Should be addressed in https://github.com/NixOS/nixpkgs/pull/219778/commits/3fa9f1f00e32bc3f90ab08c32ecb57f3ad3a05ee.

**Comment by samuela** [comment]
Looks like all checks are green now! proceeding to merge

**Pull request merged**
